utils/gridlink_impl.c.src

```diff
@@ -78,9 +78,10 @@ cellarray_DOUBLE * gridlink_DOUBLE(const int64_t NPART,
     }
 
     DOUBLE xbinsize=ZERO, ybinsize=ZERO, zbinsize=ZERO;
-    const int xstatus = get_binsize_DOUBLE(xmin,xmax,max_x_size,xbin_refine_factor, options->max_cells_per_dim, &xbinsize, &nmesh_x, options);
-    const int ystatus = get_binsize_DOUBLE(ymin,ymax,max_y_size,ybin_refine_factor, options->max_cells_per_dim, &ybinsize, &nmesh_y, options);
-    const int zstatus = get_binsize_DOUBLE(zmin,zmax,max_z_size,zbin_refine_factor, options->max_cells_per_dim, &zbinsize, &nmesh_z, options);
+
+    const int xstatus = get_binsize_DOUBLE(xdiff,max_x_size,xbin_refine_factor, options->max_cells_per_dim, &xbinsize, &nmesh_x, options);
```

Comment:
Not sure I follow the logic in this. The cellsize should be dictated by the `rpmax` and the actual spatial range of the particles. If I understand correctly, then x/y/zdiff is the periodic wrapping to employ. Since the periodic wrap must be larger than the particle extent (with no upper bound on how much larger it can be), it is possible that this calculation will hit the `max_cells_per_dim` when using the periodic wrapping distance but not when using the `xmax-xmin` separation. 

Perhaps I am missing something?